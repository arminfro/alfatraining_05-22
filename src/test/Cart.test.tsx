import userEvent from "@testing-library/user-event";
import {
  books,
  getNodeText,
  matchSnapshot,
  renderWithRouting,
  screen,
} from "./utils";

const renderCart = () =>
  renderWithRouting({ route: "/cart", store: { cart: [books[0]] } });

describe("BookDetails rendering and button clicks", () => {
  test("renders", async () => {
    renderCart();
    expect(await screen.findByTestId(/shopping-cart/i)).toBeInTheDocument();
  });

  test("matches snapshot", () => matchSnapshot(renderCart()));

  test("counter goes up", async () => {
    renderCart();
    const counter = await screen.findByTestId(/count-target/i);
    expect(getNodeText(counter)).toBe("1");
    await userEvent.click(await screen.findByTestId(/counter-up/i));
    expect(getNodeText(counter)).toBe("2");
  });

  test("counter goes down", async () => {
    renderCart();
    await userEvent.click(await screen.findByTestId(/counter-down/i));
    expect(screen.queryAllByTestId(/count-target/i).length).toBe(0);
  });
});
