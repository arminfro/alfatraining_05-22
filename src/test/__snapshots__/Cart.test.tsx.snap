// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetails rendering and button clicks matches snapshot 1`] = `
<body>
  <div>
    <ul
      class="content m-2"
      data-testid="shopping-cart"
    >
      <li>
        <a
          class="columns is-vcentered is-clickable"
          href="/books/9783864906466"
        >
          <div
            class="column is-1 m-2"
          >
            <img
              alt="Front Cover"
              class="image"
              src="https://api3.angular-buch.com/images/angular_auflage2.jpg"
            />
          </div>
          <div
            class="column is-9"
          >
            <div
              class="title is-5"
            >
              Angular
            </div>
            <div
              class="subtitle is-6 mb-2"
            >
              Grundlagen, fortgeschrittene Themen und Best Practices â€“ mit NativeScript und NgRx
            </div>
            <div
              class="has-text-grey"
            >
              <span>
                Ferdinand Malcher
                , 
              </span>
              <span>
                Johannes Hoppe
                , 
              </span>
              <span>
                Danny Koppenhagen
              </span>
              <br />
              ISBN: 
              9783864906466
            </div>
          </div>
          <div
            class="column is-2"
          >
            <div
              class="field has-addons"
            >
              <p
                class="control"
              >
                <span
                  class="button"
                  data-testid="count-target"
                >
                  1
                </span>
              </p>
              <p
                class="control"
              >
                <button
                  class="button is-success"
                  data-testid="counter-up"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="ion-add-circle-outline"
                    />
                  </span>
                </button>
              </p>
              <p
                class="control"
              >
                <button
                  class="button is-danger"
                  data-testid="counter-down"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="ion-remove-circle-outline"
                    />
                  </span>
                </button>
              </p>
            </div>
          </div>
        </a>
      </li>
    </ul>
  </div>
</body>
`;
